_posts/2020-09-10-ompl-constrained-planning-gsoc.md

```diff
@@ -0,0 +1,52 @@
+---
+author: Jeroen De Maeyer, PhD Student, KU Leuven, Belgium
+comments: false
+date: 2020-09-10
+layout: post
+title: A new approach for planning with path constraints in MoveIt
+media_type: video
+media_link: TODO(presentation PickNik)
+description: Integrating OMPL's constrained planning capabilities into MoveIt
+categories:
+- moveit
+---
+
+In the past three months, I added a new planning approach to the existing OMPL interface, during my [Google Summer of Code project](https://summerofcode.withgoogle.com/organizations/5848470994288640/#5590997586673664). It addresses known issues with the current planners to handle Cartesian path constraints, especially when the constrained region is small.
+
+The Open Motion Planning Library (OMPL) is the main library used by MoveIt to plan collision-free paths. Since 2018, OMPL included [capabilities](http://ompl.kavrakilab.org/constrainedPlanning.html) to plan paths with generic constraints represented by a function f(q) = 0, where q represents the robot’s state. These constraints are decoupled from the planning algorithms by introducing a “constrained state space”. Sampling in this constrained state space will give samples that satisfy the constraints and therefore a planner can operate as if there were no constraints. The implementation of these constrained state spaces in OMPL calculates such valid states by taking a random joint space sample and projecting it on the constraints region. This approach has some advantages compared to rejection sampling (the current approach available in MoveIt to handle arbitrary constraints). This projection sampling from OMPL is potentially faster and can handle constraints such as moving along a plane or a line, which have zero volume in Cartesian space. These constraints cannot be handled by rejection sampling.
```

Comment:
"some advantages"? Is there anything more compelling, maybe a paper to link?

I would put f(q) = 0 and q into code formatting: `f(q) = 0` etc

Reply:
I tried to make the sentence clearer and added a link to a good publication that explains all this in more detail.